CMakeLists.txt

```diff
@@ -497,14 +497,36 @@ if(BUILD_DOCS)
     install(FILES "README.html" DESTINATION "${DFHACK_DATA_DESTINATION}")
 endif()
 
-option(BUILD_TESTS "Include tests (currently just installs Lua tests into the scripts folder)" OFF)
-if(BUILD_TESTS)
-    if(EXISTS "${dfhack_SOURCE_DIR}/test/scripts")
-        message(SEND_ERROR "test/scripts must not exist in the dfhack repo since it would conflict with the tests installed from the scripts repo.")
+# Testing with CTest
+include(CTest)
+
+include(CMakeDependentOption)
+cmake_dependent_option(
+    BUILD_TEST_SCRIPTS "Install integration tests in hack/scripts/test" OFF
+    "BUILD_TESTING" OFF)
+mark_as_advanced(FORCE BUILD_TESTS)
+
+# Handle deprecated BUILD_TESTS option
+option(BUILD_TESTS "Deprecated option; please use BUILD_TEST_SCRIPTS=ON" OFF)
+if(BUILD_TESTING AND BUILD_TESTS)
+    set(BUILD_TEST_SCRIPTS ON FORCE)
+endif()
+
+if(BUILD_TESTING)
+    if(BUILD_TEST_SCRIPTS)
```

Comment:
I think the Lua test option should be independent of the c++ test option. If `BUILD_TESTING` is `OFF`, shouldn't the lua tests still be installed?